src/deploy/functions/release/executor.ts

```diff
@@ -49,8 +62,15 @@ export class QueueExecutor implements Executor {
     this.queue = new Queue({ ...options, handler });
   }
 
-  async run<T>(func: () => Promise<T>): Promise<T> {
-    const op: Operation = { func };
+  async run<T>(func: () => Promise<T>, opts?: RunOptions): Promise<T> {
+    // merge and de-duplicate default and provided retry codes
+    let retryCodes = [...defaultRetryCodes, ...(opts?.retryCodes || [])];
```

Comment:
I wonder if it's better to make retryCodes explicit and allow people to extend DEFAULT_RETRY_CODES manually

Reply:
What do you mean by "manually"?

Reply:
OOof this has been waiting around much longer than expected. Feel free to ping me if you're ever waiting on me for more than ~1d.

I'm suggesting that it's probably a more typical design that `opts.retryCodes` is literally the only retry codes that will be retried. If I was a consumer of the API and said `retryCodes: [420]` but was also getting retries on 429 I'd be pretty surprised and frustrated. Rather, make `DEFAULT_RETRY_CODES` a public part of the API and make `opts.retryCodes` completely override `DEFAULT_RETRY_CODES`. Then, if I wanted to handle all defaults _and_ 420, I'd use `retryCodes: [...DEFAULT_RETRY_CODES, 420]`

Reply:
Got it. Today, the executor retries on 429, 409, and 503 error codes for every API call wrapped in `run(...)`, and I would assume that we would like to avoid breaking existing behavior that may rely on retrying the default codes. So most executor runs would need to be edited to `.run(async () => {...}, { retryCodes: DEFAULT_RETRY_CODES })` which may actually turn out to be a non-trivial re-write.

Reply:
I'm trying to say that if {retryCodes} isn't specified, it should default to DEFAULT_RETRY_CODES. If {retryCodes} is specified, we should only retry on the specified codes. It follows the principle of least surprises to only retry on the codes which we were told to retry or have sane defaults when unspecified.

Reply:
Ah I see what you mean — definitely agree with this approach, next commit implements the behavior you're describing.